### PR TITLE
Include extras require in metadata extraction

### PR DESCRIPTION
**Pull request by mcyprian**
Including extras require listed in setup.py in build requires can prevent tests of doc building failures in some cases. PR also contains updated integration tests data.

### CODE REVIEW

**Comment by irushchyshyn** [comment]
The changes look reasonable to me. Are there a lot of cases which fail due to missing extras_require?
Is it intended only for `SetupPyMetadataExtractor`? Should not it work the same way when extracting matadata from wheel archive?

File: command/extract_dist.py [hunk 1]
```diff
@@ -22,6 +22,14 @@ def __init__(self, *args, **kwargs):
                      'packages', 'py_modules', 'scripts']:
             self.metadata[attr] = to_list(getattr(self.distribution, attr, []))
 
+        try:
+            self.metadata['extras_require'] = []
+            for v in getattr(self.distribution, 'extras_require', {}).values():
+                self.metadata['extras_require'] += to_list(v)
+        except AttributeError:
```
Comment (irushchyshyn) [inline command/extract_dist.py]: Should not we also catch `ValueError` here (which can be raised in `to_list`) or is it a rare case?

Comment (mcyprian) [reply command/extract_dist.py]: Right, I will add it.

**Comment by mcyprian** [comment]
Maybe it is not obvious, but extras are already supported in wheel extractor. Format of metadata is a bit different in metadata.json file, "extra" is only a special tag of package included in dependencies.

**Comment by mcyprian** [comment]
I added a trivial logic to distinguish between runtime and build extra requirements.

**Comment by irushchyshyn** [comment]
Yeah, I did not see that it is already supported in wheel extractor.
Looks good!

**Pull request merged**